projects/packages/external-connections/src/class-external-connections.php

```diff
@@ -0,0 +1,315 @@
+				return null;
+			}
+			$body = json_decode( wp_remote_retrieve_body( $response ) );
+			if ( isset( $body->connections ) && is_array( $body->connections ) ) {
+				foreach ( $body->connections as $connection ) {
+					if ( $service === $connection->service ) {
+						return (array) $connection;
+					}
+				}
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	 * Checks if a connection to the specified service exists and is active.
+	 *
+	 * @param string $service The service identifier.
+	 *
+	 * @return bool True if an active connection exists, false otherwise.
+	 */
+	public static function has_connection( $service ) {
+		$connection = self::get_connection( $service );
+
+		if ( empty( $connection ) ) {
+			return false;
+		}
+
+		$connection_status = $connection['status'] ?? 'not_connected';
+		return $connection_status === 'ok';
+	}
+
+	/**
+	 * Deletes a connection for the provided service.
+	 *
+	 * @param string $service The service identifier.
+	 * @return boolean Whether the connection was deleted.
+	 */
+	public static function delete_connection( $service ) {
+		$connection = self::get_connection( $service );
+		if ( empty( $connection ) ) {
+			return false;
+		}
+
+		if ( ( new Host() )->is_wpcom_simple() ) {
+			if ( get_current_user_id() === $connection['user_ID'] ) {
+				require_lib( 'external-connections' );
+				$connections = \WPCOM_External_Connections::init();
+				return $connections->delete_keyring_connection( $connection['ID'] );
+			}
+			return false;
+		}
+
+		$response = Client::wpcom_json_api_request_as_user(
+			'/me/connections/' . $connection['ID'],
+			'2',
+			array( 'method' => 'DELETE' )
+		);
+		if ( is_wp_error( $response ) ) {
+			return false;
+		}
+		$body = json_decode( wp_remote_retrieve_body( $response ) );
+		return $body->deleted ?? false;
+	}
+
+	/**
+	 * Registers connection settings.
+	 */
+	public static function register_settings() {
+		foreach ( self::$services as $page => $services ) {
+			global $pagenow;
+
+			if ( $pagenow !== "options-$page.php" ) {
+				continue;
+			}
+
+			$host = new Host();
+
+			if ( ! $host->is_wpcom_simple() ) {
+				$connection = new Connection_Manager( 'jetpack' );
+				$status     = new Status();
+
+				if ( $status->is_offline_mode() || ! $connection->has_connected_owner() || ! $connection->is_user_connected() ) {
+					return;
+				}
+			}
+
+			add_settings_section(
+				'jetpack_external_connections_section',
+				__( 'Integrations', 'jetpack-external-connections' ),
+				'__return_false',
+				$page
+			);
+
+			$asset_name = 'jetpack-external-connections-settings';
+			Assets::register_script(
+				$asset_name,
+				"build/$asset_name/$asset_name.js",
+				self::BASE_FILE,
+				array(
+					'in_footer'    => true,
+					'textdomain'   => 'jetpack-external-connections',
+					'dependencies' => array( 'wp-util' ),
+				)
+			);
+			Assets::enqueue_script( $asset_name );
+
+			$script_data = array();
+
+			foreach ( $services as $service ) {
```

Comment:
Hmmm... the $services variable comes from the foreach. I think this might introduce a subtle issue: if the $pagenow doesn't have a key, it will default to the last element in the self::$services array, right? Which means services from the last page of the array will register on pages that don't have any services? 🤔 

It's an untested theory. WDYT?